lib/buffer.js

```diff
@@ -1071,6 +1120,72 @@ Buffer.prototype.writeDoubleBE = function writeDoubleBE(val, offset, noAssert) {
   return offset + 8;
 };
 
+
+function checkInt64(buffer, value, offset, ext) {
+  if (!(buffer instanceof Buffer))
+    throw new TypeError('buffer must be a Buffer instance');
+  if (!(typeof value === 'string' || typeof value === 'number'))
+    throw new TypeError('must pass a "string" or "number" for value');
+  if (offset + ext > buffer.length)
+    throw new RangeError('index out of range');
+}
+
+
+Buffer.prototype.writeInt64LE = function writeInt64LE(val, offset, noAssert) {
+  offset = offset >>> 0;
+  if (!noAssert)
+    checkInt64(this, val, offset, 8);
+  return binding.writeInt64LE(this, val, offset);
+};
+
+
+Buffer.prototype.writeInt64BE = function writeInt64BE(val, offset, noAssert) {
+  offset = offset >>> 0;
+  if (!noAssert)
+    checkInt64(this, val, offset, 8);
+  return binding.writeInt64BE(this, val, offset);
```

Comment:
For all these `write*64*()` operations, you should check if `val` is a number and if it's within the 2^51 range that can be handled by JS. If that's the case then just do the bit twiddling here. Just an idea. If you don't implement it then I will in my rewrite. :)